specification/common/attribute-precedence.md

```diff
@@ -0,0 +1,199 @@
+```
+
+#### Span links
+
+```
+Span.links.attributes > Span.attributes > ScopeSpans.scope.attributes > ResourceSpans.resource.attributes
+```
+
+### Metrics
+
+Metrics are different from Spans and LogRecords, as each Metric has a data field
+which can contain one or more data points.
+Each data point has a set of attributes, which need to be considered 
+independently.
+
+```
+Metric.data.data_points.attributes > ScopeMetrics.scope.attributes > ResourceMetrics.resource.attributes
+```
+
+#### Metric exemplars
+
+```
+Metric.data.data_points.exemplars.filtered_attributes > Metric.data.data_points.attributes > ScopeMetrics.scope.attributes > ResourceMetrics.resource.attributes
+```
+
+### Logs
+
+```
+LogRecord.log_records.attributes > ScopeLogs.scope.attributes > ResourceLogs.resource.attributes
+```
+
+## Considerations
+
+Note that this precedence is a strong suggestion, not a requirement.
+Code that transforms attributes should follow this mode of flattening, but might 
+diverge if they have a reason to do so. 
+Furthermore, exporters can apply clash prevention, e.g., by prefixing all 
+Resource attributes with `resource.`.
+Note that even then, a Span/Metric data point/LogRecord attribute can overwrite
+the resource attribute `attribute_name`, if it is called 
+`resource.attribute_name`.
+Therefore, extra care needs to be taken when prefixing attributes.
+
+## Example
+
+De-duplication can be thought of as a map with unique keys to which the 
+attributes are added, from most general to most specialized.
+First, the resource attributes are added, then the InstrumentationLibraryScope 
+attributes, which overwrite the resource attributes if they share a key.
+Then the attributes on the Span/Metric data point/LogRecord are added, which
+again overwrite keys that are already present.
+The final set of key-value pairs are all the pairs in the map.
+
+This YAML-like representation of a theoretical OTLP message has attributes
+with attribute names that clash on multiple levels.
+
+```yaml
+ResourceMetrics:
+    resource:
+        attributes:
+            # key-value pairs (attributes) on the resource
+            attribute1: resource-attribute-1
+            attribute2: resource-attribute-2
+            attribute3: resource-attribute-3
+            service.name: my-service
+
+    scope_metrics:
+        scope:
+            attributes:
+                attribute1: scope-attribute-1
+                attribute2: scope-attribute-2
+        
+        metrics:
+            # there can be multiple data entries here.
+            data/0:
+                data_points:
+                    # each data can have multiple data points:
+                    data_point/0:
+                        attributes: 
+                            attribute1: data-point-0-attribute-1
+
+                    data_point/1:
+                        attributes: 
+                            attribute1: data-point-1-attribute-1
```

Comment:
Maybe this one can be `attribute2` to show a data point overriding a scope attribute?

Reply:
It does overwrite the scope attribute, scope also sets `attribute1`, or am i missing something?